elyra/contents/file_parser.py

```diff
@@ -105,7 +106,11 @@ def __init__(self, operation_filepath):
 
         with open(self.operation_filepath) as f:
             self.notebook = nbformat.read(f, as_version=4)
-            language = self.notebook['metadata']['kernelspec']['language']
+
+            try:
+                language = self.notebook['metadata']['kernelspec']['language']
+            except KeyError:
+                raise KeyError(f'No language metadata found in {self._operation_filepath}.')
 
             if language == 'python':
```

Comment:
The `language` comparisons should probably be case-insensitive.  The R notebooks I have show the language as 'R'.

Reply:
Great point!